### ISSUE TITLE
Fix Validate by Domain

### ISSUE THREAD

**Issue by alex-418** [BCcampus/eypd#687]
Thanks @paulagaube @dshaykewich  for the troubleshooting during out stand-up. I've renamed the plugin via https://github.com/BCcampus/validate-by-domain/commit/d92f2bbf38ef8a414ccb4c65809c33ba926adb87 and have also confirmed there's no validation happening due to the double email opt-in work that we did in https://github.com/BCcampus/eypd/pull/670/files to make users enter their e-mail twice.

I've created this issue to unpack what work is involved to make Validate by Domain work with the new field. It might be quick and just need to update the field name/value in the settings. But it may not be that simple as both of these do their own validation which could conflict.

**Comment by paulagaube** [comment]
@alex-418, FYI, the Validate by Domain plugin seemed to be working as expected on eypd.bccampus.ca when I tested it about an hour ago. i.e., My new sign up account was not validated as an Organizer because my email domain name was not on the whitelist for the plugin.

**Comment by alex-418** [comment]
@paulagaube thanks, I made some fixes but could not verify if they made it over to cert due to VPN being down for me. Can you try it again now?

**Comment by paulagaube** [comment]
I will try again now!

**Comment by paulagaube** [comment]
@alex-418 Validate by Domain is NOT working now. Both Kelsey and I tried to sign up as an Organizer, but our role did not change to Editor.

**Comment by alex-418** [comment]
@paulagaube Thanks, looks like the changes haven't made it over. The admin screen should say "field name" instead of number.

**Comment by dshaykewich** [comment]
@alex-418 please take the following actions:
1. Remove the confirm email address field
2. Ensure that the validate by domain settings point to the correct field and works properly.

**Comment by paulagaube** [comment]
@dshaykewich why do you want Alex to remove the confirm email address field? Just to see if that is causing a conflict?

**Comment by dshaykewich** [comment]
@paulagaube the two different email validations are being done in two different ways in two different places.  This will simplify the validation.

**Comment by alex-418** [comment]
@paulagaube @kkilbey This is now resolved on cert and ready for validation, please visit https://eypd.bccampus.ca/sign-up/ and confirm there's only one e-mail field and that the "whitelist" is working when trying to register as an organizer. 

Thanks @dshaykewich for running the build jobs on cert, we'll have to do that again for prod once you're ready to make these changes live.

**Comment by kkilbey** [comment]
@alex-418 I just signed up on eypd.bccampus.ca as an Organizer, but my role did not change to Editor when I activated my account.

**Comment by alex-418** [comment]
@paulagaube @kkilbey @dshaykewich I've just pushed these changes to prod and as I suspected, they are working as expected there! The validation whitelist works on the e-mail field for organizers, and the correct role is assigned when organizer account is activated. (I temporarily whitelisted my personal email domain to test this). So, this in pretty good shape :) 

Please let me know if you encounter any other issues.

**Comment by kkilbey** [comment]
Tested on prod and found the following things working as expected:
- there is only one email field on the sign-up page
- I could not sign up with a hotmail email address, but I successfully signed up with my BCcampus email address
- my role changed to "Editor" once I activated my account

@alex-418 Just to confirm, my role still didn't change when I tried signing up as an Organizer on the cert site. Is that expected behaviour for now? It's just working on prod?

@paulagaube Leaving this in Validating so you can test, too. :)

**Comment by alex-418** [comment]
@kkilbey Perfect, thank you. Yes, cert won't work as expected with the role assignment until #677 is completed.

**Comment by alex-418** [comment]
> @alex-418 please take the following actions:
> 
> 1. Remove the confirm email address field
> 2. Ensure that the validate by domain settings point to the correct field and works properly.

After our discussion yesterday at the stand-up, I wanted to give you guys a better breakdown of the options for adding another email confirmation field. @dshaykewich @paulagaube @kkilbey 

**Option 1:** Integrate with the BuddyPress & Validate by Domain
Here's a visual of what the flow looks like in [WordPress registration hooks](https://usersinsights.com/wordpress-user-registration-hooks-visualized/). BuddyPress has its own registration flow similar to this, which we currently intercept to do validation and error messages via the Validate by Domain plugin. Hooking into the existing workflow could avoid conflicts and unexpected issues. 

**Option 2:** Client-side validation
Here's a [great example](https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation) of how this works. And here's the official docs on [adding a script in WordPress](https://developer.wordpress.org/themes/basics/including-css-javascript/). I'm open to this option, and prefer it as I've done this in this theme with the Donut, Popovers, and Modals. However, these been a time sink when troubleshooting, and not always as straight forward as it seems. Also, consider the following: 

- Requires whitelist to be visible on the client side and developer to update
- Removal of duplicate functionality from the Validate by Domain plugin
- It doesn't integrate with the existing BP validation errors
- Needs conditional logic on backend to load only on the registration page, and integrate with our current theme (add to functions.php, and webpack config)

To avoid exposing our whitelist, we'd have to pass user input from the script to the backend for evaluation using [wp_localize_script()](https://codex.wordpress.org/Function_Reference/wp_localize_script) making this option not so client-side anymore. 

Happy to discuss further, hope this was useful.

**Comment by paulagaube** [comment]
@alex-418 Validate by Domain is not working as expected on Cert. 

Account creation and activation works, on eypd.bccampus.ca, but my new account was not automatically made into an Organizer (with the role of Editor) as expected.
 
On eypd.bccampus.ca 
I signed up as an Organizer using a BCcampus.ca email address, 
And though an account was created, I was not authorized automatically as an Organizer/Editor.

I cleared my web cache (Chrome) before testing and also tested this on a fresh Mac using Safari. 

I have confirmed in the Dashboard > Settings > Validate by Domain that:
1. The Plugin is enabled - Enable: checked 
2. "bccampus.ca" is on the whitelist. 
3. **Field Number**: 3 *Didn't you say this should now say **Field Name**?
4. Role: Editor
